File: dnd_helper_flutter/lib/models/enums/attributes.dart
import 'package:json_annotation/json_annotation.dart';

enum Attributes {
  @JsonValue('strength')
  strength,
  @JsonValue('dexterity')
  dexterity,
  @JsonValue('constitution')
  constitution,
  @JsonValue('intelligence')
  intelligence,
  @JsonValue('wisdom')
  wisdom,
  @JsonValue('charisma')
  charisma,
}

extension AttributesExtension on Attributes {
  String get value {
    switch (this) {
      case Attributes.strength:
        return 'strength';
      case Attributes.dexterity:
        return 'dexterity';
      case Attributes.constitution:
        return 'constitution';
      case Attributes.intelligence:
        return 'intelligence';
      case Attributes.wisdom:
        return 'wisdom';
      case Attributes.charisma:
        return 'charisma';
    }
  }
}

/// Converts to and from [Attributes] and [String]>.
class AttributesConverter implements JsonConverter<Attributes?, String?> {
  /// Create a new instance of [Attributes].
  const AttributesConverter();

  @override
  Attributes? fromJson(String? json) {
    if (json == null) return null;
    return Attributes.values
        .firstWhere((e) => e.toString().split('.').last == json);
  }

  @override
  String? toJson(Attributes? object) {
    return object?.toString().split('.').last;
  }
}

File: dnd_helper_backend/dnd_helper_backend.Core/ValueObjects/Attributes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace dnd_helper_backend.Core.ValueObjects
{
    public class Attributes
    {
        public Attributes() { }
        public Attributes(int strength, int dexterity, int constitution, int intelligence, int wisdom, int charisma) 
        {
            Strength = strength;
            Dexterity = dexterity;
            Constitution = constitution;
            Intelligence = intelligence;
            Wisdom = wisdom;
            Charisma = charisma;
        }

        public int Strength {  get; set; }
        public int Dexterity {  get; set; }
        public int Constitution {  get; set; }
        public int Intelligence {  get; set; }
        public int Wisdom {  get; set; }
        public int Charisma {  get; set; }
    }
}
